src/sinks/clickhouse.rs

```diff
@@ -76,8 +76,13 @@ impl SinkConfig for ClickhouseConfig {
         let tls_settings = TlsSettings::from_options(&self.tls)?;
         let client = HttpClient::new(tls_settings)?;
 
+        let mut config = self.clone();
+        let uri = self.endpoint.parse::<UriSerde>()?;
```

Comment:
Is there a reason we do the parsing down here rather than making `endpoint` `UriSerde` on the `ClickhouseConfig`?